src/model/app.rs

```diff
@@ -56,4 +103,58 @@ impl App {
     pub fn move_cursor_down(&mut self) {
         self.buffer.move_cursor_down();
     }
+
+    pub fn save_file(&mut self) {
+        assert!(self.app_mode() == AppMode::Command(CommandMode::Write));
+
+        let file_path = self.get_command_buffer_text_as_iter().join("");
```

Comment:
I feel like you could have just made another method at the app level that gives the text from the buffer just as is. Rather than in a vector. This all seems rather unnecessary

Reply:
Right! I resolved this in https://github.com/eldon-chung/raccoon-editor/pull/25/commits/1546a4d8756586461eab1bab4c87c4e2d096ae01.